### PR TITLE
Extensible set of unmanaged calling conventions

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
Incorporate feedback from https://github.com/dotnet/coreclr/pull/23974#issuecomment-482991750

cc @jkoritzinsky @weltkante @tannergooding

File: proposals/function-pointers.md [hunk 1]
```diff
@@ -235,6 +235,23 @@ void* v = &CloseHandle;
...
+### Extensible set of unmanaged calling conventions
+
+The set of unmanaged calling conventions supported by the current ECMA-335 encodings is outdated. We have seen requests to add support
+for more unmanaged calling conventions, for example:
+
+- [vectorcall](https://docs.microsoft.com/en-us/cpp/cpp/vectorcall) https://github.com/dotnet/coreclr/issues/12120
+- StdCall with explicit this https://github.com/dotnet/coreclr/pull/23974#issuecomment-482991750
+
+The design of this feature should allow extending the set of unmanaged calling conventions as needed in future. The problems include
+limitted space for encoding calling conventions (12 out of 16 values are taken in `IMAGE_CEE_CS_CALLCONV_MASK`) and number of places
```
Comment (AaronRobinsonMSFT) [inline proposals/function-pointers.md]: ~~limitted~~ ->limited

File: proposals/function-pointers.md [hunk 2]
```diff
@@ -235,6 +235,23 @@ void* v = &CloseHandle;
...
+that need to be touched in order to add a new calling convention. A potential solution is to introduce a new encoding that represents
+the calling convention using `System.Runtime.InteropServices.CallingConvention` enum.
```
Comment (AaronRobinsonMSFT) [inline proposals/function-pointers.md]: Please embed the link to the official documentation for the enum - https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.callingconvention

File: proposals/function-pointers.md [hunk 3]
```diff
@@ -235,6 +235,23 @@ void* v = &CloseHandle;
 func* cdecl bool(IntPtr) f1 = (func* cdecl bool(IntPtr))v;
 ```
 
+### Extensible set of unmanaged calling conventions
+
+The set of unmanaged calling conventions supported by the current ECMA-335 encodings is outdated. We have seen requests to add support
+for more unmanaged calling conventions, for example:
+
+- [vectorcall](https://docs.microsoft.com/en-us/cpp/cpp/vectorcall) https://github.com/dotnet/coreclr/issues/12120
```
Comment (AaronRobinsonMSFT) [inline proposals/function-pointers.md]: ~~https://docs.microsoft.com/en-us/cpp/cpp/vectorcall~~ -> https://docs.microsoft.com/cpp/cpp/vectorcall

When providing a link, the locale should be removed from the URL so when navigation occurs the current browser locale is respected.

**Pull request merged**